src/pages/resultsView/comparison/ResultsViewComparisonUtils.ts

```diff
@@ -1,6 +1,168 @@
-import { ComparisonGroup } from '../../groupComparison/GroupComparisonUtils';
+import {
+    ComparisonGroup,
+    finalizeStudiesAttr,
+    getNumPatients,
+    getNumSamples,
+} from '../../groupComparison/GroupComparisonUtils';
+import { getStudiesAttr } from '../../groupComparison/comparisonGroupManager/ComparisonGroupManagerUtils';
+import { Sample } from '../../../shared/api/generated/CBioPortalAPI';
+import { IQueriedMergedTrackCaseData } from '../ResultsViewPageStore';
+import {
+    isMergedTrackFilter,
+    parseOQLQuery,
+    UnflattenedOQLLineFilterOutput,
+} from '../../../shared/lib/oql/oqlfilter';
+import { SessionGroupData } from '../../../shared/api/ComparisonGroupClient';
+import ComplexKeyMap from '../../../shared/lib/complexKeyDataStructures/ComplexKeyMap';
+import { SingleGeneQuery } from '../../../shared/lib/oql/oql-parser';
+import oql_parser from '../../../shared/lib/oql/oql-parser';
+import _ from 'lodash';
 
 export type ResultsViewComparisonGroup = ComparisonGroup & {
     nameOfEnrichmentDirection: string;
     count: number;
 };
+
+export const ALTERED_COLOR = '#dc3912';
+export const UNALTERED_COLOR = '#3366cc';
+export const ALTERED_GROUP_NAME = 'Altered group';
+export const UNALTERED_GROUP_NAME = 'Unaltered group';
+
+export function completeSessionGroups(
+    patientLevel: boolean,
+    groups: SessionGroupData[],
+    sampleSet: ComplexKeyMap<Sample>,
+    getColor: () => string
+) {
+    // (1) ensure color and other required members
+    // (2) filter out, and add list of, nonexistent samples
+    // (3) add patients
+
+    return groups.map(groupData => {
+        // assign color to group if no color given
+        let color = groupData.color || getColor();
+
+        const { nonExistentSamples, studies } = finalizeStudiesAttr(
+            groupData,
+            sampleSet
+        );
+
+        return Object.assign({}, groupData, {
+            color,
+            studies,
+            nonExistentSamples,
+            uid: groupData.name,
+            nameWithOrdinal: groupData.name,
+            ordinal: '',
+            nameOfEnrichmentDirection: '',
+            count: patientLevel
+                ? getNumPatients({ studies })
+                : getNumSamples({ studies }),
+        });
+    });
+}
+
+export function getAlteredVsUnalteredGroups(
+    patientLevel: boolean,
+    studyIds: string[],
+    alteredSamples: Sample[],
+    unalteredSamples: Sample[]
+): SessionGroupData[] {
+    return [
+        {
+            name: ALTERED_GROUP_NAME,
+            description: `Number (percentage) of ${
+                patientLevel ? 'patients' : 'samples'
+            } that have alterations in the query gene(s) that also have a deep deletion in the listed gene.`,
```

Comment:
not sure if this is existing language,  but might be more clear:  "that have BOTH alteration in query query gene and deep deletion in listed gene."

Reply:
this is existing language, but if you think we should change it we can loop back pretty easily. what do you think, @tmazor ?